src/plcc/load_spec/parse_spec/parse_semantic_spec/parse_semantic_spec_test.py

```diff
@@ -0,0 +1,53 @@
+from pytest import raises
+from .parse_semantic_spec import parse_semantic_spec, parse_codeFragments, DividerMustBeginSectionError
+from plcc.load_spec.load_rough_spec.parse_lines import Line, parse_lines
+from plcc.load_spec.load_rough_spec.parse_blocks import Block
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+
+def test_basic():
+    lines_divider_and_blocks = [divider_for_tests, codeFragment_line_for_tests, block_for_tests1]
+    semantic_spec = parse_semantic_spec(lines_divider_and_blocks)
+    assert semantic_spec.language, semantic_spec.tool == 'Java'
+    assert semantic_spec.codeFragmentList == parse_codeFragments(lines_divider_and_blocks[1:])
+
+def test_empty_list():
+    with raises(IndexError):
+        parse_semantic_spec([])
+
+def test_Divider_not_first_raise_error():
+    lines_divider_and_blocks = [
+        Line('# Comment', 1, None), divider_for_tests, codeFragment_line_for_tests, block_for_tests1]
+    with raises(DividerMustBeginSectionError):
+        parse_semantic_spec(lines_divider_and_blocks)
+
+def test_no_CodeFragments():
+    list_with_Divider = [divider_for_tests]
+    semantic_spec = parse_semantic_spec(list_with_Divider)
+    assert semantic_spec.codeFragmentList == []
+    assert semantic_spec.language, semantic_spec.tool == 'Java'
+
+def test_multiple_dividers_raise_error():
+    lines_divider_and_blocks = [
+        divider_for_tests, codeFragment_line_for_tests,
+        block_for_tests1, divider_for_tests]
+    with raises(DividerMustBeginSectionError):
+        parse_semantic_spec(lines_divider_and_blocks)
+
+def test_divider_missing_raise_error():
+    lines_and_blocks = [codeFragment_line_for_tests, block_for_tests1]
+    with raises(DividerMustBeginSectionError):
+        parse_semantic_spec(lines_and_blocks)
+
+def make_block():
+    return  Block(list(parse_lines('''\
+%%%
+block
+%%%
+''')))
+
+divider_for_tests = Divider(tool='Java', language='Java', line=Line('%', 1, None))
+
+codeFragment_line_for_tests = Line('Class:init', 1, None)
+
+block_for_tests1 = make_block()
+block_for_tests2 = make_block()
```

Comment:
Similar to above, having these global variables be methods would be more useful, that way you only define objects such as a Line in one place, as you are still defining a Line object above, even though you have this here. You can have parameters for these to make them more versatile, as the design you currently have requires you to instantiate a new object if it differs slightly from your pre-defined object variables.